shared/feature/camera/src/main/java/com/epmedu/animeal/camera/presentation/CameraView.kt

```diff
@@ -0,0 +1,136 @@
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.sharp.Lens
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.core.content.ContextCompat
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import java.io.File
+import java.util.concurrent.Executor
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+@Composable
+fun CameraView(
+    onImageCaptured: (Uri) -> Unit,
+    onError: (ImageCaptureException) -> Unit,
+    onBackPressed: () -> Unit
+) {
+
+    BackHandler(enabled = true) {
+        onBackPressed()
+    }
+
+    val lensFacing = CameraSelector.LENS_FACING_BACK
+    val context = LocalContext.current
+    val lifecycleOwner = LocalLifecycleOwner.current
+
+    val coroutineScope = rememberCoroutineScope()
+
+    val preview = Preview.Builder().build()
+    val previewView = remember { PreviewView(context) }
+    val imageCapture: ImageCapture = remember { ImageCapture.Builder().build() }
+    imageCapture.targetRotation = Surface.ROTATION_90
+    val cameraSelector = CameraSelector.Builder()
+        .requireLensFacing(lensFacing)
+        .build()
+
+    LaunchedEffect(lensFacing) {
+        val cameraProvider = context.getCameraProvider()
+        cameraProvider.unbindAll()
+        cameraProvider.bindToLifecycle(
+            lifecycleOwner,
+            cameraSelector,
+            preview,
+            imageCapture
+        )
+
+        preview.setSurfaceProvider(previewView.surfaceProvider)
+    }
+
+    Box(contentAlignment = Alignment.BottomCenter, modifier = Modifier.fillMaxSize()) {
+        AndroidView({ previewView }, modifier = Modifier.fillMaxSize())
+
+        IconButton(
+            modifier = Modifier.padding(bottom = 20.dp),
+            onClick = {
+                coroutineScope.launch(Dispatchers.IO) {
+                    imageCapture.takePhoto(
+                        executor = ContextCompat.getMainExecutor(context),
+                        onImageCaptured = onImageCaptured,
+                        onError = onError
+                    )
+                }
+            },
+            content = {
+                Icon(
+                    imageVector = Icons.Sharp.Lens,
+                    contentDescription = "Take picture button",
```

Comment:
Do we have an agreement to avoid using the hardcoded text?

Reply:
Indeed, thanks. For now I replaced it on "null", as done in other places in the code.